workspaces/src/rpc/client.rs

```diff
@@ -46,15 +46,20 @@ pub struct Client {
 }
 
 impl Client {
-    pub(crate) fn new(rpc_addr: &str) -> Self {
+    pub(crate) fn new(rpc_addr: &str) -> Result<Self> {
         let connector = JsonRpcClient::new_client();
-        let rpc_client = connector.connect(rpc_addr);
+        let mut rpc_client = connector.connect(rpc_addr);
+        if let Ok(api_key) = std::env::var("NEAR_RPC_API_KEY") {
```

Comment:
Using an env feels very much like a downstream feature, we should probably just accept API key input

```rust
let mut worker = workspaces::mainnet().await?;
worker.set_api_key(..)?;
```

Which then sets the header for the inner `NetworkClient`.

Reply:
This was mainly done for the security reason that workspaces would most likely be run in CI and could potentially expose these keys if people weren't careful about it. But then again, there might be cases where its easier to set this programmatically. I'll add it, but will need a warning not to use this in a public facing CI

Reply:
That makes sense, in that case the env var is sufficient. Was thinking of workspaces from the context of a client library, much like `near-api-js`, whereas it's much more than that and the target scope is mostly for testing, esp since, by default, it installs a sandbox. All good, then, feel free to skip over this comment.